### PR TITLE
Adds year to calibration_year_index to have 21 total years of data

### PR DESCRIPTION
**Pull request by ErinCain**
I changed the year range of year options to 1979-2000 since this is what we started doing in cache-habitat (Do we want to be using this 1979-2000 or 1979-1999 or 1980-200? It seems like we are not always consistent throughout the other packages) 

I changed the year range of target years to 1998-2018 to get a full 21 year range. 

I was not able to add a year to the grand tab data because there wasn't an additional year column in the csv we are using to generate grand tab values.

### CODE REVIEW

**Comment by thesadie** [comment]
It should target 1997-2017 to accommodate spawning for 1998

**Comment by thesadie** [comment]
we don't need an additional grandtab year either

File: R/set_synth_years.R [hunk 1]
```diff
@@ -1,58 +1,61 @@
...
+  params$proportion_diverted <- params$proportion_diverted[ , , years]
+  params$total_diverted <- params$proportion_diverted[ , , years]
+  params$delta_proportion_diverted <- params$delta_proportion_diverted[ , years, ]
+  params$delta_total_diverted <- params$delta_total_diverted[ , years, ]
+  params$prop_flow_natal <- params$prop_flow_natal[ , spawn_years]
+  params$upper_sacramento_flows <- params$upper_sacramento_flows[ , years]
+  params$delta_inflow <- params$delta_inflow[ , years, ]
+  params$proportion_flow_bypass <- params$proportion_flow_bypass[ , years, ]
+  params$gates_overtopped <- params$gates_overtopped[ , years, ]
+  params$vernalis_temps <- params$vernalis_temps[ , years] # should this have spawn_years
```
Comment (thesadie) [inline R/set_synth_years.R:45]: @ergz and @ErinCain for the few inputs that I have the comment `# should this have spawn_years`, can you plz confirm what years it should be. The originals start 1979-2000, but should they be 1980-2000? Or does the indexing need to be adjusted in the model where they are used? For proxy should they be 1997-2017 or 1998-2017?

**Comment by thesadie** [comment]
@ErinCain I made a fair bit of changes to `set_synth_years` and modified the proxy year index script to make sure the indices were not out of bounds. Plz take a look

File: R/set_synth_years.R (hunk 1 above)
Comment (ErinCain) [reply R/set_synth_years.R:45]: Looks like `vernalis_temps`, `prisoners_point_temps`, `avg_temp`, and  `avg_temp_delta` are only used in rearing survival and migratory survival functions (these dynamics are all months 1-8) so we should not need the `spawn_years` for these

Comment (thesadie) [reply R/set_synth_years.R]: thanks for updating the temperature package to remove 1979 data where it was not applicable I removed these comments.

**Comment by thesadie** [comment]
@ErinCain after we release `DSMhabitat` post deer creek update, let's rerun the data caches here and then merge to main.

**Pull request merged**